src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -41,12 +41,12 @@ public NuGetFramework(string framework, Version version)
         /// <summary>
         /// Creates a new NuGetFramework instance, with an optional profile (only available for netframework)
         /// </summary>
-        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string frameworkProfile)
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string? frameworkProfile)
             : this(frameworkIdentifier, frameworkVersion, profile: ProcessProfile(frameworkProfile), platform: string.Empty, platformVersion: FrameworkConstants.EmptyVersion)
```

Comment:
I would vote to get rid of `ProcessProfile()`
```suggestion
            : this(frameworkIdentifier, frameworkVersion, profile: frameworkProfile ?? string.Empty, platform: string.Empty, platformVersion: FrameworkConstants.EmptyVersion)
```